File: V1/Examples/PHP/extracting-managing-ocr/cloud-storage/extract-ocr-or-hocr-text-from-image.php
//ExStart:
<?php
require_once realpath(__DIR__ . '/../..') . '/vendor/autoload.php';
require_once realpath(__DIR__ . '/../..') . '/Utils.php';

use Aspose\OCR\OcrApi;
use Aspose\OCR\AsposeApp;

class ExtractText {

	public $barcodeApi;

	public function __construct() {
		AsposeApp::$appSID = Utils::appSID;
		AsposeApp::$apiKey = Utils::apiKey;
		$this->ocrApi = new OcrApi();
	}

	public function extractTextFromImages() {
		//Set the image file name
		$name = "Sampleocr.bmp";

		//Set the language of the document
		$language = "English";

		//Set X and Y coordinate to recognize text inside
		$rectX = null;
		$rectY = null;

		//Set Width and Height to recognize text inside
		$rectWidth = null;
		$rectHeight = null;

		//Set the spelling correction is used
		$useDefaultDictionaries = true;

		// Set 3rd party cloud storage server (if any)
		$storage = "";

		// Set folder location at cloud storage
		$folder = "";

		// upload file to aspose cloud storage
		Utils::uploadFile($name);
			
		//invoke Aspose.OCR Cloud SDK API to extract text and partsinfo from an image
		$response = $this->ocrApi->GetRecognizeDocument($name, $language, $rectX, $rectY, $rectWidth, $rectHeight, $useDefaultDictionaries, $storage, $folder);
		print_r ( $response );
	}
}

$extractText = new ExtractText();
$extractText->extractTextFromImages();

?>
//ExEnd:

File: V1/Examples/Python/extracting-managing-ocr/cloud-storage/extract-ocr-or-hocr-text-from-image.py
import asposeocrcloud
from asposeocrcloud.OcrApi import OcrApi
from asposeocrcloud.OcrApi import ApiException

import asposestoragecloud 
from asposestoragecloud.StorageApi import StorageApi
from asposestoragecloud.StorageApi import ResponseMessage

import ConfigParser

config = ConfigParser.ConfigParser()
config.readfp(open(r'../../data/config.properties'))
apiKey = config.get('AppConfig', 'api_key')
appSid = config.get('AppConfig', 'app_sid')
out_folder = config.get('AppConfig', 'out_folder')
data_folder = "../../data/" #resouece data folder

#ExStart:1
#Instantiate Aspose.Storage API SDK
storage_apiClient = asposestoragecloud.ApiClient.ApiClient(apiKey, appSid, False)
storageApi = StorageApi(storage_apiClient)

#Instantiate Aspose.OCR API SDK
api_client = asposeocrcloud.ApiClient.ApiClient(apiKey, appSid, False)
ocrApi = OcrApi(api_client);

#Set the image file name
name = 'Sampleocr.bmp';

#Set the language of the document
language = 'English';

#Set the spelling correction is used
useDefaultDictionaries = 'true';

try:
    #upload file to aspose cloud storage
    response = storageApi.PutCreate(name, data_folder + name)
    
    #invoke Aspose.OCR Cloud SDK API to extract text and partsinfo from an image
    response = ocrApi.GetRecognizeDocument(name, language=language,useDefaultDictionaries=useDefaultDictionaries )
    
    if response.Status == 'OK':
        print "Text :: " + response.Text     
        
        print "\n\nExtract OCR or HOCR Text from Images, Done!"
                        
except ApiException as ex:
            print "ApiException:"
            print "Code:" + str(ex.code)
            print "Message:" + ex.message    
#ExEnd:1    